### ISSUE TITLE
no-noninteractive-tabindex should not fire if the value of role is an expression

### ISSUE THREAD

**Issue by jessebeach** [evcohen/eslint-plugin-jsx-a11y#589]
This should pass

```
<div
  role={role}
  tabIndex={tabindex}
/>
```

**Comment by ljharb** [comment]
Why?

If the goal of the rule is to have certainty about a11y, then dynamic values should be forbidden by default.

**Comment by jessebeach** [comment]
The goal is to deliver certainty when we can. It's unreasonable to forbid dynamic values. Dynamic values will need to be checked at runtime. Linting isn't a solution that provides 100% coverage; it's just a layer of protection.

**Comment by ljharb** [comment]
We forbid dynamic values of `button` type attributes by default - I don't find it unreasonable.

I agree it doesn't have to provide 100% coverage - that's what override commands and options are for.

**Comment by jessebeach** [comment]
@ljharb I recommend we eliminate the ambiguous signal first. Options to make the linter more strict can be layered in later.

**Comment by ljharb** [comment]
My preference is that the defaults should always be strictest, and options used to loosen.

**Comment by jessebeach** [comment]
@ljharb except the reality I'm seeing in my codebase is that developers are putting suppression comments for this case. I don't want to train developers to just suppress these rules because they lose trust in their quality. Strict rules in an ambiguous case just create mistrust and developers will learn to ignore the rule entirely.

**Comment by jessebeach** [comment]
Addressed in #606
